### PR TITLE
perimeter weights, fix kernel, set_ops

### PR DESCRIPTION
**Pull request by martinfleis**
- revise perimeter weights
- fix kernel weirdess
- start working on set ops

I started doing set ops but got stuck on the expected logic there that would not break our sorting and sparse I/O assumptions. Will need to discuss that later today. See the comment in `intersection` in the meantime.

### CODE REVIEW

File: libpysal/graph/_set_ops.py [hunk 1]
```diff
@@ -34,46 +34,42 @@ def __xor__(self, other):
...
     return False
-    return not left.adjacency.join(
-        right.adjacency, on=("focal", "neighbor"), how="inner"
-    ).empty
 
 
 def intersection(left, right):
     """
-    Keep only links that are in both left and right Graph objects.
+    Returns a binary weights object, that includes only those neighbor pairs that exist
```
Comment (jGaboardi) [inline libpysal/graph/_set_ops.py:58]: This should be `Graph`, correct?

```suggestion
    Returns a binary Graph object, that includes only those neighbor pairs that exist
```

**Comment by martinfleis** [comment]
The failures are interesting as I don't see them locally. We may need to revise how the perimeter weighting works in relation to floating point precision.

File: libpysal/graph/_utils.py [hunk 2]
```diff
@@ -167,105 +167,6 @@ def _validate_sparse_input(sparse, ids=None):
     ), "coordinates should represent a distance matrix if metric='precomputed'"
     return _sparse_to_arrays(sparse, ids)
 
-
-def lat2Graph(nrows=5, ncols=5, rook=True, id_type="int"):
```
Comment (jGaboardi) [inline libpysal/graph/_utils.py:171]: What's the reasoning behind removing `lat2Graph()`?

Comment (martinfleis) [reply libpysal/graph/_utils.py:171]: this one doesn't work :D

Comment (jGaboardi) [reply libpysal/graph/_utils.py:171]: Welp... that's a good reason.

**Comment by martinfleis** [comment]
This is ready as is for now. I would merge this and then merge geographs to main, then follow in a sequence of smaller targeted PRs based on the checklist in #562.

**Comment by ljwolf** [comment]
I can review in the next hour or so

**Review by jGaboardi — APPROVED** [review]

**Review by ljwolf — APPROVED** [review]

**Pull request merged**